src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -369,6 +376,13 @@ abstract class AbstractCheckTask(
                 .build())
         } finally {
             RuleMetric.decrementActiveRule(type())
+
+            // We need to convert event to protobuf here because
+            // getAggregatedStatus() method is 'protected' in
+            // com.exactpro.th2.common.event.Event class
+            val status = refs.rootEvent.toProto(null).status
+            result.complete(status)
```

Comment:
I think we can change the `publishEvent` to return aggregated event status (it produces batches and we can check them)